store/src/java/com/zimbra/cs/service/FileUploadServlet.java

```diff
@@ -180,6 +188,27 @@
                 if (contentType == null)
                     contentType = MimeConstants.CT_APPLICATION_OCTET_STREAM;
             }
+
+            Account acct = Provisioning.getInstance().getAccount(acctId);
+            String extension = FilenameUtils.getExtension(filename).trim();
+            String [] blockedFileTypes = null;
+
+            if (acct.getCOS().isFeatureFileTypeUploadRestrictionsEnabled()) {
+                blockedFileTypes = acct.getCOS().getMultiAttr(Provisioning.A_zimbraFileUploadBlockedFileTypes);
+                for(String blockedExt : blockedFileTypes){
+                    if(blockedExt.equalsIgnoreCase(extension)) {
+                        throw ServiceException.BLOCKED_FILE_TYPE_UPLOAD(
+                                String.format("Blocked attachment during uploading %s filetype ", blockedExt), null);
+                    }
+                }
+
+                List<String> blockedExtensionList = new ArrayList<>(Arrays.asList(blockedFileTypes));
```

Comment:
you are creating this List so that you can use blockedExtensionList.contains instead of for loop. 
You are doubling the heap use.

Reply:
you can use `Arrays.stream(arr).anyMatch()` to do `contains` check on array.

Reply:
This will not create an independent list, for the existing array we are getting the a list view. It's just a wrapper which is used to make available array available as a list and no data is copied or created so heap is not wasted.